File: algorithms/1221.split-a-string-in-balanced-strings/solution.rs
impl Solution {
    pub fn balanced_string_split(s: String) -> i32 {
        let mut res = 0;
        let mut cnt = 0;
        for c in s.chars() {
            cnt += if c == 'L' { 1 } else { -1 };
            if cnt == 0 {
                res += 1;
            }
        }
        res
    }
}

File: algorithms/1221.split-a-string-in-balanced-strings/solution.rb
# @param {String} s
# @return {Integer}
def balanced_string_split(s)
  res, cnt = 0, 0
  s.chars.each do |c|
    if c == 'L'
      cnt += 1
    else
      cnt -= 1
    end
    res += 1 if cnt == 0
  end
  res
end
